File: 1.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> index_map;
        for (int i = 0; i < nums.size(); ++i) {
            int search = target - nums[i];
            if (index_map.count(search)) {
                return vector<int>{index_map[search], i};
            } else {
                index_map[nums[i]] = i;
            }
        }
        return vector<int>();
    }
};


// ANSWER
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hashtable;
        for (int i = 0; i < nums.size(); ++i) {
            auto it = hashtable.find(target - nums[i]);
            if (it != hashtable.end()) {
                return {it->second, i};
            }
            hashtable[nums[i]] = i;
        }
        return {};
    }
};

File: 1.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> valMap = new HashMap<>();
        for (int i = 0; i < nums.length; ++i) {
            int num = nums[i];
            if (valMap.containsKey(target - num)) {
                return new int[] { valMap.get(target - num), i };
            }
            valMap.put(nums[i], i);
        }
        return new int[] {};
    }
}